src/app/services/views/ServiceDataPlaneProxiesView.vue

```diff
@@ -32,7 +32,7 @@
           :src="`/meshes/${route.params.mesh}/dataplanes/for/${route.params.service}/of/${'all'}?page=${route.params.page}&size=${route.params.size}&search=${route.params.s}`"
         >
           <template
-            v-for="gateways in [typeof dataplanesData?.items?.[0]?.dataplane?.networking?.gateway === 'undefined']"
+            v-for="gateways in [dataplanesData?.items?.[0]?.dataplane?.networking?.gateway !== undefined]"
```

Comment:
Feels like this is one of those PRs where we fix the bug (and add tests to catch in future if appropriate, which you've done) and nothing more, we can then push anything else separately.

Just so I'm clear this line here is the bug fix right? Is the rest (in `DataPlaneList.vue`) necessary to fix the bug?

Reply:
The other changes are more important in the sense that they’ll allow the links for the rows to always be correct independent from this check.

Removing this dependency from the value created here makes _this_ fix less important, but still relevant. It’s still not great that we decide the entire table is either gateway or DPP by just looking at the first element, but at least it now influences fewer things.

Reply:
ok gotcha, thanks, 2 tics.